File: fizzbuzz.hs
-- FizzBuzz in Haskell

fizzBuzz n | n `mod` 15 == 0 = "FizzBuzz"
	   | n `mod` 3  == 0 = "Fizz"
	   | n `mod` 5  == 0 = "Buzz"
	   | otherwise       = show n

main = do
    putStr $ show $ map fizzBuzz [1..100]

File: fizzbuzz.rb
for i in 1..100
    case i % 15
    when 0
	puts 'fizzbuzz'
    when 3, 6, 9, 12
	puts 'fizz'
    when 5, 10
	puts 'buzz'
    else
	puts i
    end
end

block = lambda {|n|
    case i % 15
    when 0
	puts 'fizzbuzz'
    when 3, 6, 9, 12
	puts 'fizz'
    when 5, 10
	puts 'buzz'
    else
	puts i
    end
}

100.times block
